aea/helpers/cid.py

```diff
@@ -0,0 +1,277 @@
+
+        if version not in (0, 1):
+            raise ValueError(
+                "version should be 0 or 1, {} was provided".format(version)
+            )
+
+        if not multicodec.is_codec(codec):
+            raise ValueError("invalid codec {} provided, please check".format(codec))
+
+        if not isinstance(multihash, bytes):
+            raise ValueError("invalid type for multihash provided, should be bytes")
+
+        if version == 1:
+            return CIDv1(codec, multihash)
+
+        if codec != CIDv0.CODEC:
+            raise ValueError(
+                "codec for version 0 can only be {}, found: {}".format(
+                    CIDv0.CODEC, codec
+                )
+            )
+        return CIDv0(multihash)
+
+    @classmethod
+    def is_cid(cls, cid: str) -> bool:
+        """Checks if a given input string is valid encoded CID or not."""
+
+        try:
+            return bool(cls.from_string(cid))
+        except ValueError:
+            return False
+
+    @classmethod
+    def from_string(cls, cid: str) -> CIDObject:
+        """Creates a CID object from a encoded form"""
+
+        cid_bytes = ensure_bytes(cid, "utf-8")
+        return cls.from_bytes(cid_bytes)
+
+    @classmethod
+    def from_bytes(cls, cid: bytes) -> CIDObject:
+        """Creates a CID object from a encoded form"""
+
+        if len(cid) < 2:
+            raise ValueError("argument length can not be zero")
+
+        # first byte for identity multibase and CIDv0 is 0x00
+        # putting in assumption that multibase for CIDv0 can not be identity
+        # refer: https://github.com/ipld/cid/issues/13#issuecomment-326490275
+
+        if cid[0] != 0 and multibase.is_encoded(cid):
+            # if the bytestream is multibase encoded
+            cid = multibase.decode(cid)
+            if len(cid) < 2:
+                raise ValueError("cid length is invalid")
+
+            data = cid[1:]
+            version = int(cid[0])
+            codec = multicodec.get_codec(data)
+            multihash = multicodec.remove_prefix(data)
+
+        elif cid[0] in (0, 1):
+            # if the bytestream is a CID
+            version = cid[0]
+            data = cid[1:]
+            codec = multicodec.get_codec(data)
+            multihash = multicodec.remove_prefix(data)
+
+        else:
+            # otherwise its just base58-encoded multihash
+            try:
+                version = 0
+                codec = CIDv0.CODEC
+                multihash = base58.b58decode(cid)
+            except ValueError:
+                raise ValueError("multihash is not a valid base58 encoded multihash")
+
+        mh.decode(multihash)
+        return cls.make(version, codec, multihash)
+
+
+def to_v0(hash_string: str) -> str:
+    """Convert CID v1 hash to CID v0"""
+
+    cid = CID.from_string(hash_string)
+    if cid.version == 0:
+        raise ValueError(f"{hash_string} is already v0.")
+
+    return cast(CIDv1, cid).to_v0().encode().decode()
```

Comment:
add encoding parameter as in `to_v1`, it is also used in `CIDv0.encode`

Reply:
v0 used `base58` encoding by default and I think it only supports one encoding format so we don't need encoding parameter here.